janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void graphConfigurationShouldBeWhatWeExpect() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
+            assertEquals(graph.getConfiguration().getConfiguration().get(STORAGE_BACKEND), "inmemory");
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void shouldCreateAndGetGraphUsingTemplateConfiguration() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            final StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+
+            assertNotNull(graph);
+            assertEquals(graph, graph1);
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void graphConfigurtionShouldBeWhatWeExpectWhenUsingTemplateConfiguration()
+        throws InterruptedException, BackendException, Exception {
+
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            final StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+
+            assertNotNull(graph);
+            assertEquals(graph, graph1);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
```

Comment:
Expected value (constant) is first argument to `assertEquals()`. Same for the next line as well.